karma.conf.js

```diff
@@ -0,0 +1,133 @@
+// Karma configuration
+// Generated on Fri Sep 25 2020 11:02:54 GMT+0200 (Central European Summer Time)
+
+const { join: joinPath } = require( 'path' );
+
+const basePath = process.cwd();
+const coverageDir = joinPath( basePath, 'coverage' );
+
+module.exports = function(config) {
+	config.set({
+
+		// base path that will be used to resolve all patterns (eg. files, exclude)
+		basePath,
+
+
+		// frameworks to use
+		// available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+		frameworks: ['mocha', 'chai', 'sinon'],
+
+
+		// list of files / patterns to load in the browser
+		files: [
+			'tests/**/*.js'
+		],
+
+
+		// preprocess matching files before serving them to the browser
+		// available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+		preprocessors: {
+			'tests/**/*.js': [ 'webpack' ]
+		},
+
+
+		webpack: {
+			mode: 'development',
+			devtool: 'inline-source-map',
+
+			module: {
+				rules: [
+					{
+						test: /\.js$/,
+						loader: 'babel-loader',
+						exclude: /node_modules/,
+						query: {
+							compact: false,
+							presets: [ '@babel/preset-env' ]
+						}
+					},
+					{
+						test: /\.js$/,
+						loader: 'istanbul-instrumenter-loader',
+						include: /src/,
+						exclude: [
+							/node_modules/
+						],
+						query: {
+							esModules: true
+						}
+					}
+				]
+			}
+		},
+
+		webpackMiddleware: {
+			noInfo: true,
+			stats: 'minimal'
+		},
+
+
+		// test results reporter to use
+		// possible values: 'dots', 'progress'
+		// available reporters: https://npmjs.org/browse/keyword/karma-reporter
+		reporters: [
+			'mocha',
+			'coverage'
+		],
+
+
+		coverageReporter: {
+			reporters: [
+				// Prints a table after tests result.
+				{
+					type: 'text'
+				},
+				// Generates HTML tables with the results.
+				{
+					dir: coverageDir,
+					type: 'html'
+				},
+				// Generates "lcov.info" file. It's used by external code coverage services.
+				{
+					type: 'lcovonly',
+					dir: coverageDir
+				}
+			]
+		},
+
+
+		// web server port
+		port: 9876,
+
+
+		// enable / disable colors in the output (reporters and logs)
+		colors: true,
+
+
+		// level of logging
+		// possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
+		logLevel: config.LOG_INFO,
+
+
+		// enable / disable watching file and executing tests whenever any file changes
+		autoWatch: false,
+
+
+		// start these browsers
+		// available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+		browsers: ['ChromeHeadless'],
+
+
+		// Continuous Integration mode
+		// if true, Karma captures browsers, runs the tests and exits
+		singleRun: true,
```

Comment:
The option we have in [Angular repo](https://github.com/ckeditor/ckeditor4-angular/blob/master/src/karma.conf.js#L30) to set this to `false` from console is pretty useful for me, maybe we could introduce it here too - WDYT? It would also require changing the `autoWatch` to `true` and parsing the options, but it can [just be copied](https://github.com/ckeditor/ckeditor4-angular/blob/master/src/karma.conf.js#L4-L5https://github.com/ckeditor/ckeditor4-angular/blob/master/src/karma.conf.js#L4-L5).

Reply:
I don't think there we need to go with such complicated options handling for that. I will just add a separate node script for that (`npm run test-watch`).